qemu/target-arm/translate.c

```diff
@@ -10408,15 +10407,11 @@ static void disas_thumb_insn(CPUARMState *env, DisasContext *s) // qq
     }
 
     // Unicorn: trace this instruction on request
-    if (env->uc->hook_insn) {
-        struct hook_struct *trace = hook_find(s->uc, UC_HOOK_CODE, s->pc);
-        if (trace)
-            gen_uc_tracecode(tcg_ctx, 2, trace->callback, env->uc, s->pc, trace->user_data);
-        // if requested to emulate only some instructions, check to see
-        // if we need to exit immediately
-        if (env->uc->emu_count > 0) {
-            check_exit_request(tcg_ctx);
-        }
+    struct hook *hook;
```

Comment:
we are following C89, so please declare all the variables at the beginning of block